diesel/Cargo.toml

```diff
@@ -38,6 +38,7 @@ version = "~2.0.0"
 path = "../diesel_derives"
 
 [dev-dependencies]
+assert_matches = "1.0.1"
```

Comment:
I would prefer not to add another dependency here, especially a simple on like that one.

This macro + a ordinary `assert!` should work well enough for that test case.
```rust
macro_rules! matches {
    ($expression:expr, $( $pattern:pat )|+ $( if $guard: expr )?) => {
        match $expression {
            $( $pattern )|+ $( if $guard )? => true,
            _ => false
        }
    }
}
```